projects/plugins/jetpack/modules/carousel/jetpack-carousel.php

```diff
@@ -817,7 +817,10 @@ class_exists( 'Jetpack_AMP_Support' )
 	 * @param string $content HTML content of the post.
 	 * @return string
 	 */
-	public function add_data_img_tags_and_enqueue_assets( $content ) {
+	public function add_data_img_tags_and_enqueue_assets( $content = '' ) {
+		if ( ! is_string( $content ) ) {
+			$content = '';
```

Comment:
What about returning `$content` here? If it's not a string we won't be able to do much with that content anyway, we could bail early that way?

```suggestion
			return $content;
```

Reply:
Sure, though I'll return an empty string, as that's the expected return type: 08b4f44971

Reply:
I worry that fixing the return type would hide errors introduced elsewhere, either in Jetpack or in other plugins. If something filters `the_content` and returns something wrong, I think the site owner should know about it.

Reply:
On the other hand, the input and output of this function, as well as the input of `the_content` filter, should always be a string. If we were in control of the entire flow, I'd prefer returning `null`, but in this case it just kicks the problem down the road to another (perhaps less robust third-party) hook. Normalizing the data to an expected type is less likely to cause errors.

I don't have a strong opinion, though, so if you prefer returning `null`, I'll change it. :^)

Reply:
I don't have that strong of an opinion on this, so I'm good with returning an empty string as well, that's not a blocker for me :)